app/workers/extract_fulltext_job.rb

```diff
@@ -0,0 +1,80 @@
+#-- encoding: UTF-8
+
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2017 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2017 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+class ExtractFulltextJob < ApplicationJob
+  def initialize(attachment_id)
+    @attachment_id = attachment_id
+    @attachment = nil
+    @text = nil
+    @file = nil
+    @filename = nil
+    @language = OpenProject::Configuration.main_content_language
```

Comment:
I can unfortunately no longer recall what we decided regarding having the language as a configuration as opposed to having it as a setting. The later would allow the user to alter it via the UI but might require us to rerun the ts_vector rake task upon changes to it. For SaaS however, this will probably be necessary as english might not always be the correct setting. I therefore think it is "alternativlos"
 
![image](https://user-images.githubusercontent.com/617519/36250156-46a2cb58-123d-11e8-82c6-a85003cd6d7c.png)

Reply:
We don't communicate it and that is already a problem. E.g. the rake task in this PR will suffer from the same problem already as we provide no feedback there as well. E-mail sending would be another example where it would make sense to provide feedback. But as we already have that problem I wouldn't, having the same problem again for reindexing does not sound too bad to me.

But maybe it is ok to roll with what we have (others have done so too: https://github.com/zotonic/zotonic/commit/c8aad274489029dc5d8928c6781034253ca9d00e) and see how far we can get. 

But let's keep the topic in the back of our heads once additional features like trigrams are added which will rely on the language more heavily.

Reply:
@ulferts I am not sure if understand everything.

Yes, the rake tasks could be more informative. But I did not understand the rest of your explanation. Let's quickly catch up on this on Monday.